mmdet/models/detectors/d2_wrapper.py

```diff
@@ -0,0 +1,306 @@
+                as `gt_instance` or `gt_panoptic_seg` or `gt_sem_seg`.
+
+        Returns:
+            dict: A dictionary of loss components.
+        """
+        d2_batched_inputs = self._convert_to_batched_d2_inputs(
+            batch_inputs=batch_inputs,
+            batch_data_samples=batch_data_samples,
+            training=True)
+
+        with self.storage as storage:  # noqa
+            losses = self.d2_model(d2_batched_inputs)
+        # storage contains some training information, such as cls_accuracy.
+        # you can use storage.latest() to get the detail information
+        return losses
+
+    def predict(self, batch_inputs: Tensor,
+                batch_data_samples: SampleList) -> SampleList:
+        """Predict results from a batch of inputs and data samples with post-
+        processing.
+
+        The inputs will first convert to the Detectron2 type and feed into
+        D2 models. And the results will convert back to the MMDet type.
+
+        Args:
+            batch_inputs (Tensor): Input images of shape (N, C, H, W).
+                These should usually be mean centered and std scaled.
+            batch_data_samples (list[:obj:`DetDataSample`]): The batch
+                data samples. It usually includes information such
+                as `gt_instance` or `gt_panoptic_seg` or `gt_sem_seg`.
+
+
+        Returns:
+            list[:obj:`DetDataSample`]: Detection results of the
+            input images. Each DetDataSample usually contain
+            'pred_instances'. And the ``pred_instances`` usually
+            contains following keys.
+
+            - scores (Tensor): Classification scores, has a shape
+              (num_instance, )
+            - labels (Tensor): Labels of bboxes, has a shape
+              (num_instances, ).
+            - bboxes (Tensor): Has a shape (num_instances, 4),
+              the last dimension 4 arrange as (x1, y1, x2, y2).
+        """
+        d2_batched_inputs = self._convert_to_batched_d2_inputs(
+            batch_inputs=batch_inputs,
+            batch_data_samples=batch_data_samples,
+            training=False)
+        # results in detectron2 has already rescale
+        d2_results_list = self.d2_model(d2_batched_inputs)
+        batch_data_samples = add_d2_pred_to_datasample(
+            data_samples=batch_data_samples, d2_results_list=d2_results_list)
+
+        return batch_data_samples
+
+    def _forward(self, *args, **kwargs):
+        """Network forward process.
+
+        Usually includes backbone, neck and head forward without any post-
+        processing.
+        """
+        raise NotImplementedError(
+            f'`_forward` is not implemented in {self.__class__.__name__}')
+
+    def extract_feat(self, *args, **kwargs):
+        """Extract features from images.
+
+        `extract_feat` will not be used in obj:``Detectron2Wrapper``.
+        """
+        pass
+
+    def _convert_to_batched_d2_inputs(self,
```

Comment:
It seems that the returned list is still a list, so is it better to change the name to `_convert_to_d2_inputs`?